constexpr_string.hpp

```diff
@@ -17,35 +17,39 @@ class constexpr_string
     using const_iterator = const char*;
 
 public:
-    constexpr constexpr_string(): data_{}, size_{N}
```

Comment:
I liked short constructor initializers list on same line as constructor, it also complied with Google style that we use at work, I'll see on Monday our file at work!
https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists